swig/lightgbmlib.i

```diff
@@ -74,41 +74,48 @@
     return dst;
   }
 
-  int LGBM_BoosterPredictForMatSingle(JNIEnv *jenv,
-                                      jdoubleArray data,
-                                      BoosterHandle handle,
-                                      int data_type,
-                                      int ncol,
-                                      int is_row_major,
-                                      int predict_type,
-                                      int num_iteration,
-                                      const char* parameter,
-                                      int64_t* out_len,
-                                      double* out_result) {
+  jdoubleArray LGBM_BoosterPredictForMatSingle(JNIEnv *jenv,
+                                               jdoubleArray data,
+                                               BoosterHandle handle,
+                                               int data_type,
+                                               int ncol,
+                                               int is_row_major,
+                                               int predict_type,
+                                               int num_iteration,
+                                               const char* parameter,
+                                               int64_t* out_len) {
     double* data0 = (double*)jenv->GetPrimitiveArrayCritical(data, 0);
+    // Note: we allocate the output array in the native side but return new java array
+    double* out_result = new double[*out_len];
 
     int ret = LGBM_BoosterPredictForMatSingleRow(handle, data0, data_type, ncol, is_row_major, predict_type,
                                                  num_iteration, parameter, out_len, out_result);
 
     jenv->ReleasePrimitiveArrayCritical(data, data0, JNI_ABORT);
+    
+    if (ret != 0) {
+      return nullptr;
```

Comment:
looks like I need to delete before returning here

delete[] out_result;

Reply:
can you use 

```
vector<double> data0;
data0.resize(*out_len);
```

this will get you RAII behavior for errors (exceptions,...)

Reply:
great idea